### PR TITLE
DietPi-Backup | Backup archive and lifted /mnt limit

### LINKED ISSUE #1785: DietPi-Backup | Backup archive with selectable max backups
Ref: https://dietpi.com/phpbb/viewtopic.php?t=3593

### LINKED ISSUE #2886: DietPi-Backup | Allow backup targets outside of /mnt/ and add to exclude list automatically
## Non standard locations (non:/mnt/) for backups are needed

I encountered an issue where a previous system labeled all the large media devices and now, I just cannot backup due to this. Is there a work around or do I need to make this feature request and work one myself?

### PR DESCRIPTION
**Pull request by MichaIng**
**Status**: Ready

**Commit list/description**:
DietPi-Backup | A backup archive with a selectable amount of backups to keep can be created now. Backups are rotated automatically and if the maximum amount has been reached, the oldest backup is used as basis for the incremental new backup sync, to reduce writes and increase speed. Many thanks to @johnvick and many others for requesting this feature: https://dietpi.com/phpbb/viewtopic.php?t=3593

### CODE REVIEW

**Comment by MichaIng** [comment]
Works pretty well here, though it at least one limitation which we might want to address: If a backup fails, this does not prevent the rotation, hence e.g. 10 backups are configured + daily backups enabled, if failures remain unrecognised 10 days, all backups could be broken. The logic to prevent this is however not that easy. We could rotate back in case of a failure, so that the last successful backup is in the first slot, but then the info from the stats and logs file does not match the backup, which may cause confusion. So better is probably to implement some notification in the banner or ultimately #2663. But I think it is acceptable since even without the archive feature, the daily cron job did overwrite the only backup and in case of failure could have broken it as well. So this feature doesn't worsen that risk but gives more reaction time, if for some reason backups start to fail, e.g. full drive or such.

I'll merge this now so it can be tested as part of the beta.

File: dietpi/dietpi-backup [hunk 1]
```diff
@@ -324,76 +338,76 @@ However, this check is a rough estimation in reasonable time, thus it could be m
 		fi
 	}
 
+	# $1: Optional restore point from the archive
 	Run_Restore(){
 
+		local DATA='data'
+		(( $1 > 1 )) && DATA="data_$1"
```
Comment (Joulinar) [inline dietpi/dietpi-backup:345]: this line is giving an error on restore now

```
root@DietPi4:~# dietpi-backup -1
/boot/dietpi/dietpi-backup: line 345: ((: > 1 : syntax error: operand expected (error token is "> 1 ")
```

Comment (MichaIng) [reply dietpi/dietpi-backup:345]: Ah true, in case you do not explicitly restore on of the older backups. It is however visual only. I'll mute it.

Comment (MichaIng) [reply dietpi/dietpi-backup:345]: Done: https://github.com/MichaIng/DietPi/commit/fbc02d6

**Pull request merged**